packages/instant/src/types.ts

```diff
@@ -0,0 +1,9 @@
+export enum ActionTypes {
+    UPDATE_ETH_USD_PRICE,
+    UPDATE_SELECTED_ASSET_AMOUNT,
+}
+
+export interface Action {
+    type: ActionTypes;
+    data?: any;
```

Comment:
I'm not that familiar with redux, so excuse the ignorance here, but I'm wondering: but does this mean that you can trigger an ActionType with any type of `data`?  If so, is there any way to be more specific, to only allow you to only send trigger certain actions with certain data types?

Reply:
Good point -- I mean, in non-typed environments yes data is kind of anything and then the reducer  is the one that knows that for a certain Action.type the data is going to be of a certain form.

I think it's possible with union types, so you would end up with something like:
```
type Action = BooleanAction | NumberAction ;

interface BooleanAction {
 type: ActionTypes.UpdateBooleanValueFoo | ActionTypes.UpdateBooleanValueBar;
 data: boolean 
}
```
But doesn't seem worth it?

Reply:
There might be some way to do it with generics that doesn't drastically increase the boilerplate.

I skimmed thru some of this: https://medium.com/@martin_hotell/improved-redux-type-safety-with-typescript-2-8-2c11a8062575

I'm fine with merging as is, especially because we already use this convention in website and it hasn't caused any problems, even with all the different action types there

Reply:
I love typesafety, so to mean it seems worth it, but I'm cool with rolling as-is for now if that's what ya'll would like to do: we could potentially revisit later once we have more actions

Reply:
Lets look into this later - I agree that having types here would be great, but would like to avoid boilerplate if possible. Seems like a bit of rabbit hole and I think this branch is blocking a few things so going to go ahead and merge it once CI passes.